ingestion_tools/dataset_configs/config-attribute-discovery.py

```diff
@@ -0,0 +1,213 @@
+    if not isinstance(current_entries[key], list):
+        current_entries[key] = [current_entries[key]]
+    return current_entries
+
+
+"""
+This function is the main function that will be called. It will take in the current dictionary and the new dictionary, and will merge them together. A recursive
+approach is used to handle nested dictionaries and lists. The function will return the merged dictionary.
+"""
+
+
+def recursive_dict_update(current_entries, new_entries):
+    for key, new_value in new_entries.items():
+        # Regular scenarios
+        if new_value is None:
+            continue
+        # Dict and dict, so recurse
+        if isinstance(new_value, dict) and (
+            isinstance(current_entries.get(key), dict) or current_entries.get(key) is None
+        ):
+            current_entries[key] = recursive_dict_update(current_entries.get(key, {}), new_value)
+        # List and list situation
+        elif isinstance(new_value, list) and (
+            isinstance(current_entries.get(key), list) or current_entries.get(key) is None
+        ):
+            current_entries = recursive_dict_update_list_helper(current_entries, key, new_value)
+        # Dict and non-dict
+        elif not isinstance(new_value, dict) and isinstance(current_entries.get(key), dict):
+            # edge case: dict and a list, add the dict to the list
+            if isinstance(new_value, list):
+                new_list = new_value + [current_entries.get(key)]
+                # and then now it is list, list situation
+                current_entries = recursive_dict_update_list_helper(current_entries, key, new_list)
+            else:
+                print("dict conflict")
+                print("Key: ", key)
+                print("Current: ", current_entries.get(key))
+                print("New: ", new_value)
+        # List and non-list
+        elif not isinstance(new_value, list) and isinstance(current_entries.get(key), list):
+            # edge case: list and a dict, add the dict to the list
+            if isinstance(new_value, dict):
+                new_list = [new_value] + current_entries.get(key)
+                # and then now it is list, list situation
+                current_entries = recursive_dict_update_list_helper(current_entries, key, new_list)
+            # edge case: one-length list and a non-list, non-dict new_value = just keep the list
+            elif (
+                len(current_entries.get(key)) == 1
+                and not isinstance(new_value, list)
+                and not isinstance(new_value, dict)
+            ):
+                pass
+            else:
+                print("list conflict")
+                print("Key: ", key)
+                print("Current: ", current_entries.get(key))
+                print("New: ", new_value)
+        else:
+            raw_data_check(current_entries.get(key, None), new_value)
+            current_entries[key] = new_value
```

Comment:
What are your thoughts on capturing the types of values passed for this field?

Reply:
Done!